src/main/java/org/broadinstitute/consent/http/resources/VoteResource.java

```diff
@@ -0,0 +1,102 @@
+package org.broadinstitute.consent.http.resources;
+
+
+import com.google.gson.Gson;
+import com.google.gson.reflect.TypeToken;
+import io.dropwizard.auth.Auth;
+import org.broadinstitute.consent.http.enumeration.ElectionStatus;
+import org.broadinstitute.consent.http.models.AuthUser;
+import org.broadinstitute.consent.http.models.Election;
+import org.broadinstitute.consent.http.models.User;
+import org.broadinstitute.consent.http.models.Vote;
+import org.broadinstitute.consent.http.service.ElectionService;
+import org.broadinstitute.consent.http.service.UserService;
+import org.broadinstitute.consent.http.service.VoteService;
+
+import javax.annotation.security.RolesAllowed;
+import javax.ws.rs.BadRequestException;
+import javax.ws.rs.Consumes;
+import javax.ws.rs.NotFoundException;
+import javax.ws.rs.PUT;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.QueryParam;
+import javax.ws.rs.core.Response;
+import java.lang.reflect.Type;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Path("api/votes")
+public class VoteResource extends Resource {
+
+    private final ElectionService electionService;
+    private final UserService userService;
+    private final VoteService voteService;
+
+    public VoteResource(ElectionService electionService, UserService userService, VoteService voteService) {
+        this.electionService = electionService;
+        this.userService = userService;
+        this.voteService = voteService;
+    }
+
+    /**
+     * This API will take a boolean vote value as a query param and apply it to the list of vote ids
+     * passed in as a list of integer vote ids.
+     *
+     * Error cases are:
+     * 1. JSON Body is not a list of integers
+     * 2. Auth user is not the owner of all votes being updated
+     * 3. No votes match the list of ids provided
+     *
+     * @param authUser The AuthUser
+     * @param vote The boolean value to update votes for
+     * @param rationale The string value for all rationales
+     * @param json The list of vote ids, in json format
+     * @return Response with results of the update.
+     */
+    @PUT
+    @Consumes("application/json")
+    @Produces("application/json")
+    @RolesAllowed({CHAIRPERSON, MEMBER})
+    public Response updateVotes(
```

Comment:
Question: What's the plan on updating the election status for a Chairperson vote? Will the front-end issue a separate back-end request to update Election, or should it be handled through this endpoint and/or its associated service methods?

Reply:
Excellent question and I think this is something I missed. I don't see a ticket for this in the epic, so will do some thinking on what to do here. I'm leaning toward updating the election based on the types of votes passed in.

Reply:
It's the only gap in the PR right now. Functionally the API so far makes sense, and the process checks to see if the election is open before updating the votes (so a chairperson can't change their vote if the election is already closed). However if the election is open a chairperson can change their vote however many times they want.

Reply:
Addressed in the service class